hp_list/main.c

```diff
@@ -14,6 +14,120 @@
 #include <string.h>
 #include <threads.h>
 
+static atomic_uint_fast64_t deletes = 0, inserts = 0;
+
+#ifdef RUNTIME_STAT
+/*
+ * "rtry" the number of retries in the __list_find function.
+ * "cons" the number of wait-free contains in the __list_find function that curr
+ * pointer pointed.
+ * "trav" the number of list element traversal in the __list_find function.
+ * "fail" the number of CAS() failures.
+ * "del" the number of list_delete operation failed and restart again.
+ * "ins" the number of list_insert operation failed and restart again.
+ * "load"  is the number of atomic_load operation in list_delete, list_insert
+ * and __list_find.
+ * "store" is the number of atomic_store operation in list_delete, list_insert
+ * and __list_find.
+ */
+struct runtime_stat {
+    atomic_uint_fast64_t rtry, cons, trav, fail;
+    atomic_uint_fast64_t del, ins;
+    atomic_uint_fast64_t load, store;
+};
+static struct runtime_stat stats = {0};
+
+enum {
```

Comment:
Move the enumeration before `struct runtime_stat`. And then, use inlining comments along enumeration.